libpod/runtime_ctr.go

```diff
@@ -53,6 +60,27 @@ func (r *Runtime) NewContainer(spec *spec.Spec, options ...CtrCreateOption) (c *
 		}
 	}()
 
+	shmdir := ctr.ShmDir()
+	if shmdir == "" {
+		shmdir = filepath.Join(ctr.bundlePath(), "shm")
+		if err := os.MkdirAll(shmdir, 0700); err != nil {
+			if !os.IsExist(err) {
+				return nil, errors.Wrapf(err, "unable to create shm %q dir", shmdir)
+			}
+		}
+
+		shmOptions := "mode=1777,size=" + strconv.Itoa(DefaultShmSize)
+		if err := unix.Mount("shm", shmdir, "tmpfs", unix.MS_NOEXEC|unix.MS_NOSUID|unix.MS_NODEV,
+			label.FormatMountLabel(shmOptions, ctr.config.MountLabel)); err != nil {
+			return nil, errors.Wrapf(err, "failed to mount shm tmpfs %q", shmdir)
+		}
+		ctr.config.ShmDir = shmdir
+		ctr.runtime.Mounts = append(ctr.runtime.Mounts, shmdir)
+	}
+	g := generate.NewFromSpec(spec)
+	g.AddBindMount(shmdir, "/dev/shm", []string{"rw"})
+	ctr.replaceSpec(g.Spec())
```

Comment:
Instead of using replaceSpec, can you do this in https://github.com/projectatomic/libpod/blob/master/libpod/container.go#L426-L431 - when we generate the runtime spec we're going to use? That will remove the need for replaceSpec and ensure that ctr.config.Spec is always the spec the user gave to us

Reply:
Fixed